dev/tests/integration/testsuite/Magento/Downloadable/Model/Product/TypeTest.php

```diff
@@ -250,8 +250,8 @@ public function testCheckProductBuyState()
         $linksFactory = $this->objectManager
             ->get(\Magento\Downloadable\Model\ResourceModel\Link\CollectionFactory::class);
         $allLinksIds = $linksFactory->create()->addProductToFilter($product->getEntityId())->getAllIds();
-        $this->assertEquals(
-            '{"qty":23,"links":["' . implode('","', $allLinksIds). '"]}',
+        $this->assertRegExp(
```

Comment:
We don't need to rely on concrete implementation, use `\Magento\Framework\Serialize\Serializer\Json`

Reply:
Reworked